src/containers/Node/NodeStructure/NodeStructure.tsx

```diff
@@ -0,0 +1,439 @@
+            });
+        }
+        if (valueIsDefined(SerialNumber)) {
+            pdiskInfo.push({label: 'SerialNumber', value: SerialNumber});
+        }
+        return (
+            <div>
+                <InfoViewer className={b('pdisk-details')} info={pdiskInfo} />
+                <div className={b('vdisks-container')}>
+                    <div className={b('vdisks-header')}>VDisks</div>
+                    {renderVDisks()}
+                </div>
+            </div>
+        );
+    };
+
+    return (
+        <div className={b('pdisk')} id={props.id}>
+            <div className={b('pdisk-header')}>
+                <div className={b('pdisk-title-wrapper')}>
+                    <span>{data.Path}</span>
+                    <EntityStatus status={data.Device} name={`${data.NodeId}-${data.PDiskId}`} />
+                </div>
+                <Button onClick={unfolded ? onClosePDiskDetails : onOpenPDiskDetails} view="clear">
+                    <ArrowToggle direction={unfolded ? 'top' : 'bottom'} />
+                </Button>
+            </div>
+            {unfolded && renderPDiskDetails()}
+        </div>
+    );
+}
+
+interface NodeStructureProps {
+    nodeId: string;
+    className?: string;
+    additionalNodesInfo?: any;
+}
+
+const autofetcher = new AutoFetcher();
+
+function NodeStructure(props: NodeStructureProps) {
+    const dispatch = useDispatch();
+
+    const nodeStructure: any = useSelector((state: any) => selectNodeStructure(state));
+
+    const {loadingStructure, wasLoadedStructure} = useSelector((state: any) => state.node);
+    const nodeData = useSelector((state: any) => state.node?.data?.SystemStateInfo?.[0]);
+
+    const nodeHref = useMemo(() => {
+        return props.additionalNodesInfo?.getNodeRef
+            ? props.additionalNodesInfo.getNodeRef(nodeData)
+            : undefined;
+    }, [nodeData, props.additionalNodesInfo]);
+
+    const {pdiskId: pdiskIdFromUrl, vdiskId: vdiskIdFromUrl} = url.parse(
+        window.location.href,
+        true,
+    ).query;
+
+    const isReady = useRef(false);
+
+    const scrolled = useRef(false);
+
+    const scrollContainer = useRef<Element>();
+
+    useEffect(() => {
+        return () => {
+            if (scrollContainer.current) {
+                scrollContainer.current.scrollTo({
+                    behavior: 'smooth',
+                    top: 0,
+                });
+            }
+        };
+    }, []);
+
+    useEffect(() => {
+        dispatch(getNodeStructure(props.nodeId));
+        autofetcher.start();
+        autofetcher.fetch(() => dispatch(getNodeStructure(props.nodeId)));
+
+        return () => {
+            scrolled.current = false;
+            isReady.current = false;
+            autofetcher.stop();
+        };
+    }, [props.nodeId, dispatch]);
+
+    useEffect(() => {
+        if (!_.isEmpty(nodeStructure)) {
+            isReady.current = true;
+        }
+    }, [nodeStructure]);
+
+    useEffect(() => {
```

Comment:
Could you please explain what is going on inside that effect?

Reply:
Now i receive scrollContainer as ref! It seems to be more clear way than getting element from DOM directly.

In this useEffect we should check if all elements are rendered (isReady.current is true) and scroll hasn't been done yet (scrolled.current is false) (in this page we fetch data in intervals and we don't want to scroll every time it is fetched).